.github/workflows/check-state-compatibility.yml

```diff
@@ -0,0 +1,111 @@
+
+          echo "MAINNET_MAJOR_VERSION=$MAINNET_MAJOR_VERSION" >> $GITHUB_ENV
+          echo "mainnet_major_version=$MAINNET_MAJOR_VERSION" >> $GITHUB_OUTPUT
+
+  check_mainnet:
+    # DO NOT CHANGE THIS: please read the note above
+    needs: compare_versions
+    runs-on: self-hosted
+    steps:
+      - name: 🐍 Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+
+      - name: 🐿 Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: "^${{ env.GOLANG_VERSION }}"
+
+      - name: Checkout osmosis-installer branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Checkout osmosis repo
+        uses: actions/checkout@v3
+        with:
+          repository: osmosis-labs/osmosis
+          path: repo
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: 🔨 Build the osmosisd binary
+        run: |
+          cd repo
+          make build
+          build/osmosisd version
+
+      - name: 🧪 Initialize Osmosis Node
+        run: |
+          rm -rf $HOME/.osmosisd/ || true
+          repo/build/osmosisd init runner -o
+
+          # Download genesis file if not present
+          mkdir -p /mnt/data/genesis/osmosis-1/
+          if [ ! -f "/mnt/data/genesis/osmosis-1/genesis.json" ]; then
+            wget -q -O /mnt/data/genesis/osmosis-1/genesis.json ${{ env.GENESIS_URL }}
+          fi
+
+          # Copy genesis to config folder
+          cp /mnt/data/genesis/osmosis-1/genesis.json $HOME/.osmosisd/config/genesis.json
+
+      - name: 🧪 Configure node
+        run: |
+          CONFIG_FOLDER=$HOME/.osmosisd/config
+          LAST_EPOCH_BLOCK_HEIGHT=$(curl -s --retry 5 --retry-delay 5 --connect-timeout 30 ${{ env.LCD_ENDPOINT }}/osmosis/epochs/v1beta1/epochs | dasel --plain -r json 'epochs.(identifier=day).current_epoch_start_height')
+
+          HALT_HEIGHT=$(($LAST_EPOCH_BLOCK_HEIGHT + ${{ env.DELTA_HALT_HEIGHT }}))
+
+          echo "Last Epoch Height: $LAST_EPOCH_BLOCK_HEIGHT"
+          echo "Halt Height: $HALT_HEIGHT"
+
+          # Edit config.toml
+          dasel put string -f $CONFIG_FOLDER/config.toml '.tx_index.indexer' null
+          dasel put string -f $CONFIG_FOLDER/config.toml '.p2p.persistent_peers' 'b59016320a74525f92dbc3348521c64f2bf3f006@p2p.archive.osmosis.zone:26656'
+          dasel put string -f $CONFIG_FOLDER/config.toml 'seeds' ''
+
+          # Edit app.toml
+          dasel put string -f $CONFIG_FOLDER/app.toml '.halt-height' $HALT_HEIGHT
+          dasel put string -f $CONFIG_FOLDER/app.toml '.pruning' everything
+          dasel put string -f $CONFIG_FOLDER/app.toml '.state-sync.snapshot-interval' 0
+          dasel put string -f $CONFIG_FOLDER/app.toml '.halt-time' 600
+
+          # Download addrbook
+          wget -q -O $CONFIG_FOLDER/addrbook.json ${{ env.ADDRBOOK_URL }}
```

Comment:
What is this step doing here?

This isn't configuring the end node, the installer should be doing that?

Reply:
Yes, everything should be removed except for modifying the `app.toml` to halt the chain after some blocks/time.